### ISSUE TITLE
Stability of DCAT version URIs 

### ISSUE THREAD

**Issue by NancyJean** [w3c/dxwg#1128]
My institution is investigating use of DCAT but has a very basic concern about the stability of DCAT Class and property URIs among versions.  DCAT v2 has made changes to DCAT v1 properties (e.g., loosening constraints) and Classes (e.g., changing definition of Distribution); however, the URIs for those properties and Classes do not appear to have changed from version 1 to version 2.  Hence, local models based on the v1 definitions and constraints may no longer be valid - at the very least modeling decisions made based on v1 are compromised.

Before committing to use DCAT, we need to know that if we create a data model incorporating, for example, DCAT v2, that the URIs will always represent the model defined by DCAT v2. 

Essentially, since DCAT v2 makes changes to properties and Classes that existed in DCAT v1, then DCAT v2 should have a different namespace, e.g., http://www.w3.org/ns/dcat2 and dcat2:.  And going forward, new versions should have their own unique namespaces.  This allows users stability, i.e., we don't have to review and update our local use of DCAT properties and Classes every time DCAT changes/releases a new version.  Our local builds will reference stable version URIs.

**Comment by makxdekkers** [comment]
@NancyJean We discussed the options for a new namespace for DCAT2 some time ago. See https://github.com/w3c/dxwg/issues/131.

As I see it, we are following best practice at W3C and other places where the namespace remains the same for different versions (e.g. RDF: http://www.w3.org/1999/02/22-rdf-syntax-ns#; OWL: http://www.w3.org/2002/07/owl# -- note that OWL2 still uses the OWL namespace; SKOS: http://www.w3.org/2004/02/skos/core#; Dublin Core: http://purl.org/dc/terms/; Schema.org: http://schema.org/).

Our main objective was to preserve backward compatibility as much as possible; because of that, relaxing constraints and adding new classes and properties in the existing namespace was  considered to be OK because existing implementations don't break.

DCMI has an [explicit policy](https://www.dublincore.org/specifications/dublin-core/dcmi-namespace/) for changes to URIs:
_Changes of definitions within DCMI recommendations and/or DCMI term declarations will be reflected in the affected DCMI recommendation and/or DCMI term declaration. If, in the judgment of the DCMI Directorate, such changes of meaning are likely to have substantial impact on either machine processing of DCMI terms or the functional semantics of the terms, then these changes will be reflected in a change of URI for the DCMI term or terms in question. The URIs for any new DCMI namespaces resulting from such changes will conform to the DCMI namespace URI pattern defined above._

Note the emphasis on the _substantial impact_ in this policy, and also that, if a new URI were to be created, it would still be declared in the same namespace. This is more or less the policy that we're using for DCAT. Maybe we need to make this policy explicit?

**Comment by riccardoAlbertoni** [comment]
Dear @NancyJean,
Thanks a lot for the comment.
As explained by @makxdekkers, and as documented by the discussions he pointed at, we use the same namespace as DCAT2 changes do not break the existing implementations.   DCAT is then stable. 
If you do not have other observations on this, we are going to close this issue on git. 
Is it ok for you?

**Comment by NancyJean** [comment]
I appreciate the explanation and do think that you need to make your versioning policy explicit in the documentation.  As to following the versioning policies of DCMI and W3C  - well, I suppose that's their choice as much as it is yours.

That said, I'm not entirely convinced that the reasons posited for not versioning are in the best interest of the DCAT user community.  Among the reasoning in issue 131 for not versioning are 2 seemingly self serving statements: 
"... defining a new namespace for DCAT2 should be an absolute last resort. Doing so would set the clock back in terms of establishing DCAT's acceptance - and it already has a lot of competition."

"My worry is that if this group were to define a new namespace for DCAT2 with roughly the same semantics as DCAT1, very few implementers of DCAT1 might see the need to upgrade."  

I think the latter statement makes my point -- users of DCAT1 should not have to upgrade.  As a potential DCAT user, I don't want to be forced or coerced to upgrade when a new version is issued - and really I'm concerned more about versioning changes to existing properties and Classes than the addition of new.  While changes to existing properties and Classes may not break backward compatibility, they may affect the intellectual basis of a broader model in which DCAT is just a part.  If I'm extending DCAT1 with an external property because  a narrowly defined DCAT1 property doesn't fit my needs, and then you later change/broaden the DCAT property without changing the namespace, that affects the intellectual underpinnings of my model. Agreed it doesn't necessarily break the model, but the modeling decision is no longer valid.

As to this reason - "Even if you were to define a new namespace, you'd then have the problem of people using, say, dcat:Distribution when in fact, oh silly developer, you meant dcat2:Distribution; don't you know the difference?" That seems a little flippant and I wonder if the user community is being given sufficient credit for being able to correctly apply versioned URIs.

**Comment by riccardoAlbertoni** [comment]
We have added the following note in the document section ["dcat history"](https://w3c.github.io/dxwg/dcat/) putting upfront why DCAT 2 maintains the same namespace as DCAT 2014

"DCAT 2 maintains the DCAT namespace as it preserves backward compatibility with DCAT. DCAT 2 relaxes constraints and adds new classes and properties, but these changes do not break existing DCAT implementations." 

Providing a more elaborated statement about versioning policy seems to go beyond the mere editorial changes we can do at this stage of the recommendations process. So we have moved this issue to the [milestone DCAT Future Priority Work.](  https://github.com/w3c/dxwg/milestone/15) as a reminder for the future rounds of DCAT specification.

As for the doubts that you have expressed so far, we would like to remark that DCAT works in an open-world setting. 

In particular, under the open-world assumption,
•     Domain/range restrictions are not constraints as used in the close-world system as a relational database, and eventually, they are meant to be an indication with which classes a given property should be used
•    Adding new properties to a class is meant to indicate how to specify in a harmonised way additional information. Of course, this may result in changing the actual semantics of a class, but as we are in the open-world assumption, this is part of the game.
